dashboard/utils.py

```diff
@@ -281,6 +284,21 @@ def has_passed_course(self, course_id):
                 raise ImproperlyConfigured('Missing "end_date" for course run {}'.format(course_id))
             return cur_grade.passed and course_run.is_past
 
+    def has_passed_course_for_exam(self, course_id):
+        """
+        Returns whether the user has passed a course run for an exam
```

Comment:
please add some explanation here about why we need it (meaning that for the exam purposes the `has_passed_course` that raises is equivalent to `False`